### PR TITLE
feat!: Update ESLint version requirements

### PR DESCRIPTION
**Pull request by wakamsha**
## What changed / motivation ?

To support ESLint v9, most plugins need to be updated. This will also change the minimum required version of ESLint.

## Linked PR / Issues

- https://github.com/moneyforward/frontend-forward/issues/219

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Chore (routine task, maintenance, or non-functional change that doesn't modify src or test files)

## Pre-flight checklist

- [x] I have read the [Contributing Guidelines](https://github.com/moneyforward/eslint-config-moneyforward/blob/main/.github/CONTRIBUTING.md)
- [x] Performed a self-review of my code

### CODE REVIEW

**Comment by wakamsha** [comment]
Although the snapshot test is failing on CI, the snapshot is actually updated, so please ignore this CI error 🙏

**Comment by wakamsha** [comment]
This PR is frozen as a Draft until #338 is merged.

**Comment by wakamsha** [comment]
This PR is frozen as a Draft until #355  is merged.

File: package.json [hunk 1]
```diff
@@ -55,15 +55,15 @@
     "eslint-plugin-react-hooks": "^5.0.0",
     "eslint-plugin-storybook": "^0.11.1",
     "eslint-plugin-testing-library": "^7.0.0",
-    "eslint-plugin-unicorn": "^56.0.0",
+    "eslint-plugin-unicorn": "^56.0.1",
```
Comment (wakamsha) [inline package.json:58]: When used in combination with ESLint v9.15.0 or higher, lint will fail, so you need to update eslint-plugin-unicorn.

see: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2497

**Review by taigakiyokawa — APPROVED** [review]
LGTM 🙆‍♂️ Thank you!

**Pull request merged**